### PR TITLE
fix(lint/noUnknownProperty): ignore `composes` for noUnknownProperty

### LINKED ISSUE #3000: add `composes` for noUnknownProperty.
The current implementation of `noUnknownProperty` raises an error for the composes property, which is a part of the CSS Modules syntax (refer to [CSS Modules Documentation](https://github.com/css-modules/css-modules/blob/master/docs/composition.md)). We need to modify the implementation so that it does not raise an error for the composes property if the css_modules option is enabled.

```
.className {
  color: green;
  background: red;
}

.otherClassName {
  composes: className;
  color: yellow;
}
```

### PR DESCRIPTION
**Pull request by chansuke**
## Summary

Closes #3000

### CODE REVIEW

File: crates/biome_css_analyze/src/lint/nursery/no_unknown_property.rs [hunk 1]
```diff
@@ -71,6 +71,7 @@ impl Rule for NoUnknownProperty {
         let node = ctx.query();
         let property_name = node.name().ok()?.text().to_lowercase();
         if !property_name.starts_with("--")
+            && property_name != "composes"
```
Comment (togami2864) [inline crates/biome_css_analyze/src/lint/nursery/no_unknown_property.rs:74]: I think it's better to add an explanatory comment or links

**Review by togami2864 — APPROVED** [review]

**Comment by Sec-ant** [comment]
Based on this message: https://github.com/biomejs/biome/issues/3000#issue-2319025097

> We need to modify the implementation so that it does not raise an error for the composes property ***if the css_modules option is enabled***.

I wonder if there's a way to query the option in a rule context? We can get the file source from the context but `css_modules` is not a file source variant. @denbezrukov

**Comment by denbezrukov** [comment]
> Based on this message: https://github.com/biomejs/biome/issues/3000#issue-2319025097
> 
> 
> 
> > We need to modify the implementation so that it does not raise an error for the composes property ***if the css_modules option is enabled***.
> 
> 
> 
> I wonder if there's a way to query the option in a rule context? We can get the file source from the context but `css_modules` is not a file source variant. @denbezrukov 


Hm,
I’m not familiar with the analyzer infrastructure :(
I think if it’s not possible to get the option inside the rule, we can:

	1.	Just ignore it and not use the option.
	2.	Implement an option for the rule where we can add names to be ignored and extend it if the option is enabled.

**Review by Sec-ant — APPROVED** [review]

**Comment by Sec-ant** [comment]
> I think if it’s not possible to get the option inside the rule, we can:
> 
> 1. Just ignore it and not use the option.
> 2. Implement an option for the rule where we can add names to be ignored and extend it if the option is enabled.

I think ignoring the option to always accept `composes` is fine. We can wait for more feedbacks to decide whether to add a rule option.

File: CHANGELOG.md [hunk 2]
```diff
@@ -122,6 +122,8 @@ our [guidelines for writing a good changelog entry](https://github.com/biomejs/b
 
   Contributed by @Sec-ant
 
+- The [noUnnownProperty](https://biomejs.dev/linter/rules/no-unknown-property/) rule now handles `composes` property. [#3000](https://github.com/biomejs/biome/issues/3000) Contributed by @chansuke
```
Comment (Sec-ant) [inline CHANGELOG.md:125]:
```suggestion
- The [noUnknownProperty](https://biomejs.dev/linter/rules/no-unknown-property/) rule now ignores the `composes` property often used in css modules. [#3000](https://github.com/biomejs/biome/issues/3000) Contributed by @chansuke
```

**Pull request merged**